### PR TITLE
feat(frontend): add posthog integration and cookiebot

### PR DESCRIPTION
**Pull request by DavidSpek**
WIP
## Summary
This PR is an initial setup of PostHog for the frontend. We still need a cookie banner and a way for users to opt-out of tracking. We'll probably also want to add some custom events here or there, along with expanding the user properties we set.

Todo:
- [x] Add a cookie banner and allow users to opt out of tracking
- [x] Add more properties to users instead of only their email?

## Test Plan
Run locally and check PostHog to see if users are being identified and events are being recorded as expected.

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

**Comment by DavidSpek** [comment]
Thanks for the note. I'm actually not seeing the cookie banner pop up at all so that's a bit strange.

File: www/src/components/profile/Profile.tsx [hunk 1]
```diff
@@ -44,6 +45,10 @@ export function Profile() {
   const [avatarFile, setAvatarFile] = useState<any>()
   const [mutation, { loading }] = useMutation(UPDATE_USER, {
     variables: { attributes: { name, email, avatar: avatarFile } },
+    onCompleted: ({ updateUser: { id, email } }) => {
```
Comment (michaeljguarino) [inline www/src/components/profile/Profile.tsx:48]: I'm pretty sure this is unnecessary, it'll be plumbed into the use context after update

**Review by michaeljguarino — COMMENTED** [review]
I'm not sure we need the setup in the login flows but i suppose it's not doing any damage.

We should also be positive these don't throw if posthog is down, otherwise it becomes a critical dependency on our site working.

File: www/src/components/profile/Profile.tsx (hunk 1 above)
Comment (DavidSpek) [reply www/src/components/profile/Profile.tsx:48]: Yeah that is still from before I added it in the `src/components/Plural.tsx`.

**Comment by DavidSpek** [comment]
@michaeljguarino I just check if breaking the posthog domain by adding a random path or changing the subdomain will cause issues and it doesn't from what I can tell. There's just a bunch of failed requests in the network tab and console but the app functionality is still fine including logging in.

File: www/src/components/users/MagicLogin.tsx [hunk 2]
```diff
@@ -570,6 +570,7 @@ export function Signup() {
       history.navigate('/')
     }
   }, [history])
+  // we should probably move hubspot to loaded similarly to with the other analytic tools and setup with cookiebot
   useScript({ src: 'https://js.hs-scripts.com/22363579.js' })
```
Comment (dogmar) [inline www/src/components/users/MagicLogin.tsx:574]: It was originally requested that we only needed hubspot on the login page, so I only put it here to reduce the number of places we'd need to show its consent banner, but now that we're adding a global consent banner, may as well just put it up with the rest as a regular script tag marked with `type="text/plain" data-cookieconsent="statistics"`. Also will want to add some variation on [this](https://github.com/pluralsh/webflow-helpers/blob/main/standalone/tracking-mgmt.js) to disable/re-enable hubspot when prefs change.

**Review by dogmar — CHANGES_REQUESTED** [review]

**Review by dogmar — APPROVED** [review]

**Pull request merged**